internal/core/src/mmap/Column.h

```diff
@@ -286,10 +287,108 @@ class VariableColumn : public ColumnBase {
         views_.emplace_back(data_ + indices_.back(), size_ - indices_.back());
     }
 
- private:
     std::vector<uint64_t> indices_{};
 
     // Compatible with current Span type
     std::vector<ViewType> views_{};
 };
+
+class ArrayColumn : public ColumnBase {
+ public:
+    // memory mode ctor
+    ArrayColumn(size_t num_rows,
+                const FieldMeta& field_meta,
+                const DataType element_type)
+        : ColumnBase(num_rows, field_meta), element_type_(element_type) {
+    }
+
+    // mmap mode ctor
+    ArrayColumn(const File& file,
+                size_t size,
+                const FieldMeta& field_meta,
+                const DataType element_type)
+        : ColumnBase(file, size, field_meta), element_type_(element_type) {
+    }
+
+    ArrayColumn(ArrayColumn&& column) noexcept
+        : ColumnBase(std::move(column)),
+          indices_(std::move(column.indices_)),
+          views_(std::move(column.views_)),
+          element_type_(column.element_type_) {
+    }
+
+    ~ArrayColumn() override = default;
+
+    SpanBase
+    Span() const override {
+        return SpanBase(views_.data(), views_.size(), sizeof(Array));
+    }
+
+    [[nodiscard]] const std::vector<Array>&
+    Views() const {
+        return views_;
+    }
+
+    Array
+    operator[](const int i) const {
+        return views_[i];
+    }
+
+    ScalarArray
+    RawAt(const int i) const {
+        auto array = views_[i];
+        return array.output_data();
+    }
+
+    void
+    Append(const char* data, size_t size) {
+        indices_.emplace_back(size_);
+        ColumnBase::Append(data, size);
+    }
+
+    void
+    Append(const char* data,
```

Comment:
As the array is a large type (e.g. Array of strings), providing a `Append` method that takes a `Array` would help if we move the array into the column